### PR TITLE
fix(planner): scalar subquery in function got error

### LINKED ISSUE #7250: bug: scalar subquery assertion failed
**Summary**

```
WITH
c1 as (select count() from blocks),
c2 as (select count() from transactions)
select (select * from c1)+(select * from c2);

result:
140137140636928
```

```
c1 is : 200000
c2 is: 35830797
```

explain:
```
Project: [(SELECT * FROM c1) + (SELECT * FROM c2) (#39)]
└── EvalScalar: [+(scalar_subquery_18 (#18), scalar_subquery_36 (#36))]
    └── HashJoin: SINGLE, build keys: [], probe keys: [], join filters: []
        ├── HashJoin: SINGLE, build keys: [], probe keys: [], join filters: []
        │   ├── Scan: default.system.one, filters: []
        │   └── Project: [count() (#18)]
        │       └── EvalScalar: [count() (#19)]
        │           └── Aggregate(Final): group items: [], aggregate functions: [count()]
        │               └── Aggregate(Partial): group items: [], aggregate functions: [count()]
        │                   └── Scan: default.eth.blocks, filters: []
        └── Project: [count() (#36)]
            └── EvalScalar: [count() (#37)]
                └── Aggregate(Final): group items: [], aggregate functions: [count()]
                    └── Aggregate(Partial): group items: [], aggregate functions: [count()]
                        └── Scan: default.eth.transactions, filters: []
```

### PR DESCRIPTION
**Pull request by xudong963**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Fixes #7250

### CODE REVIEW

**Review by leiysky — APPROVED** [review]

File: src/query/service/src/sql/planner/semantic/type_check.rs [hunk 1]
```diff
@@ -929,7 +930,12 @@ impl<'a> TypeChecker<'a> {
         let mut arg_types = vec![];
 
         for argument in arguments {
-            let box (arg, arg_type) = self.resolve(argument, None).await?;
+            let box (arg, mut arg_type) = self.resolve(argument, None).await?;
```
Comment (Xuanwo) [inline src/query/service/src/sql/planner/semantic/type_check.rs:933]: Do we need to handle this in `self.resolve()`?

File: src/query/service/src/sql/planner/semantic/type_check.rs [hunk 2]
```diff
@@ -929,7 +930,12 @@ impl<'a> TypeChecker<'a> {
...
+            if let Scalar::SubqueryExpr(subquery) = &arg {
```
Comment (Xuanwo) [inline src/query/service/src/sql/planner/semantic/type_check.rs:934]: Can we `if let Scalar::SubqueryExpr(subquery) = &arg && subquery.typ == SubqueryType::Scalar && !arg.data_type().is_nullable()`?

File: src/query/service/src/sql/planner/semantic/type_check.rs (hunk 1 above)
Comment (xudong963) [reply src/query/service/src/sql/planner/semantic/type_check.rs:933]: No

File: src/query/service/src/sql/planner/semantic/type_check.rs (hunk 2 above)
Comment (xudong963) [reply src/query/service/src/sql/planner/semantic/type_check.rs:934]: No, `subquery` can't be know in `subquery.typ == SubqueryType::Scalar`

**Pull request merged**